kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemDataset.java

```diff
@@ -199,11 +199,14 @@ public PathIterator pathIterator() {
     try {
       if (!fileSystem.exists(partitionDirectory)) {
         if (allowCreate) {
-          fileSystem.mkdirs(partitionDirectory);
           if (partitionListener != null) {
             partitionListener.partitionAdded(namespace, name,
                 toRelativeDirectory(key).toString());
           }
+
+          // ensure that the directory exists, it may or may not have been
+          // created by the partitionListener
+          fileSystem.mkdirs(partitionDirectory);
```

Comment:
Do we need to check to make sure it hasn't already been created or is it ok to just call mkdirs again?